### PR TITLE
Actually use our linker as the linker for sample/fastLinkJS.

### PR DESCRIPTION
**Pull request by sjrd**
~~Draft, just to get the CI to run.~~

Main changes:

1. Cross-compile the `wasm` project.
2. Turn the `Compiler` into a linker that actually implements the `Linker` and `LinkerBackend` interfaces of Scala.js.
3. Set up the `sample` project to use our WebAssembly linker in `sample/fastLinkJS` (and therefore in `sample/run`).

### CODE REVIEW

**Comment by sjrd** [comment]
Rebased on top of #50.

File: cli/src/main/scala/Main.scala [hunk 1]
```diff
@@ -45,9 +64,14 @@ object Main {
           ()
         }
       } else {
+        val linker = WebAssemblyLinkerImpl.linker(linkerConfig)
+        val outputDir = "./target/sample/"
+        createDir(outputDir)
+        val output = NodeOutputDirectory(outputDir)
```
Comment (tanishiking) [inline cli/src/main/scala/Main.scala:70]: This `else` branch is no longer needed. When we run `sample/fastLinkJS`, it's going to be handled by `WasmLInkerPlugin`.
Maybe we can remove this branch? I was a bit confused why `sample/fastLinkJS` doesn't generate output under `./target/sample`.

**Review by tanishiking — APPROVED** [review]
I left one comment, but otherwise looks good to me, great job!
feel free to merge this PR once you address the comment (if it's required)

File: cli/src/main/scala/Main.scala (hunk 1 above)
Comment (sjrd) [reply cli/src/main/scala/Main.scala]: Good point. I removed it.

**Pull request merged**